### PR TITLE
Transition group

### PR DESCRIPTION
**Pull request by jquense**

### CODE REVIEW

**Comment by jquense** [comment]
Makes our transition API compatible with the wider react community 👍 depends on the react-overlays PR tho

File: src/Modal.js [hunk 1]
```diff
@@ -128,6 +128,15 @@ const childContextTypes = {
   }),
 };
 
+/* eslint-disable no-use-before-define, react/no-multi-comp */
+const DialogTransition = (props) =>
```
Comment (taion) [inline src/Modal.js]: maybe declare these as `function DialogTransition` &c.? they're components, right?

File: src/Collapse.js [hunk 2]
```diff
@@ -114,52 +122,40 @@ const defaultProps = {
...
-  handleEnter(elem) {
-    const dimension = this._dimension();
-    elem.style[dimension] = '0';
+  handleEnter = (elem) => {
+    elem.style[this.getDimension()] = '0';
   }
 
-  handleEntering(elem) {
-    const dimension = this._dimension();
+  handleEntering = (elem) => {
```
Comment (taion) [inline src/Collapse.js]: we enabled this transform? nice

File: package.json [hunk 3]
```diff
@@ -127,8 +127,9 @@
     "keycode": "^2.1.2",
     "prop-types": "^15.5.6",
     "prop-types-extra": "^1.0.1",
-    "react-overlays": "^0.7.0",
+    "react-overlays": "^0.8.0",
     "react-prop-types": "^0.4.0",
+    "react-transition-group": "^2.0.0-beta.1",
```
Comment (taion) [inline package.json]: use final?

File: src/TabContent.js [hunk 4]
```diff
@@ -11,8 +11,8 @@ const propTypes = {
 
   /**
    * Sets a default animation strategy for all children `<TabPane>`s. Use
-   * `false` to disable, `true` to enable the default `<Fade>` animation or any
-   * `<Transition>` component.
+   * `false` to disable, `true` to enable the default `<Fade>` animation or
+   * a `react-transition-group@2.0.0` `<Transition/>` component.
```
Comment (taion) [inline src/TabContent.js]: I would just say

```
a react-transition-group v2 `<Transition>` component.
```

File: src/TabPane.js [hunk 5]
```diff
@@ -18,8 +18,8 @@ const propTypes = {
 
   /**
    * Use animation when showing or hiding `<TabPane>`s. Use `false` to disable,
-   * `true` to enable the default `<Fade>` animation or any `<Transition>`
-   * component.
+   * `true` to enable the default `<Fade>` animation or
+   * a `react-transition-group@2.0.0` `<Transition/>` component.
```
Comment (taion) [inline src/TabPane.js]: I would just say

```
a react-transition-group v2 `<Transition>` component.
```

**Comment by ujovlado** [comment]
Hi, shouldn't be this marked as breaking change? Looks like `react-transition-group` needs React `>=15.0.0`, but `react-bootstrap` still has `^0.14.9 || >=15.3.0` in peerDependencies

**Pull request merged**